src/omniperf_base.py

```diff
@@ -26,37 +26,47 @@
 import logging
 import sys
 import os
-from utils.utils import demarcate, trace_logger
-
+from pathlib import Path
+import shutil
+from utils.specs import get_machine_specs
+from utils.utils import demarcate, trace_logger, get_version, get_version_display, detect_rocprof
+from argparser import omniarg_parser
+import config
+import pyfiglet
 
 class Omniperf:
     def __init__(self):
         self.__args = None
         self.__profiler_mode = None
         self.__soc_name = None
         self.__soc = None
+        self.__version = {
+            "ver": None,
+            "ver_pretty": None,
+        }
         self.__options = {}
 
         self.setup_logging()
-        self.parseArgs()
-
-        # hard-coding dummy examples (comment/uncomment below to instantiate different implementations)
-        self.__profiler_mode = "rocprof_v1"
-        self.__profiler_mode = "rocprof_v2"
-        #self.__profiler_mode = "rocscope"
-
-        self.__analyze_mode = "cli"
-        self.__analyze_mode = "webui"
+        self.set_version()
+        self.parse_args()
 
-        # hard-code execution mode
-        self.__mode = "profile"
-        #self.__mode = "analyze"
+        self.__mode = self.__args.mode
 
+        if self.__mode == "profile":
+            self.detect_profiler()
+        
+        # self.__analyze_mode = "cli"
+        # self.__analyze_mode = "webui"
+        
         logging.info("Execution mode = %s" % self.__mode)
    
-
+    def print_graphic(self):
+        """Read program name and log ascii art to terminal.
+        """
+        ascii_art = pyfiglet.figlet_format(config.prog.capitalize())
```

Comment:
Cute but I wonder if we want to introduce another dependency (pyfiglet) just for this? We could embed the results directly perhaps.  No strong opinion, but just thinking about dependency bloat.

Reply:
Good idea. Done.